torchgeo/datasets/substation.py

```diff
@@ -0,0 +1,263 @@
+    dataset is curated by TransitionZero and sourced from publicly
+    available data repositories, including OpenSreetMap (OSMF) and
+    Copernicus Sentinel data. The dataset consists of Sentinel-2
+    images from 27k+ locations; the task is to segment power-substations,
+    which appear in the majority of locations in the dataset.
+    Most locations have 4-5 images taken at different timepoints
+    (i.e., revisits)
+
+    Dataset Format:
+    * .npz file for each datapoint
+
+    Dataset Features:
+
+    * 26,522 image-mask pairs stored as numpy files.
+    * Data from 5 revisits for most locations.
+    * Multi-temporal, multi-spectral images (13 channels) paired with masks,
+      with a spatial resolution of 228x228 pixels
+
+    If you use this dataset in your research, please cite the following:
+
+    * https://doi.org/10.48550/arXiv.2409.17363
+    """
+
+    directory = 'Substation'
+    filename_images = 'image_stack.tar.gz'
+    filename_masks = 'mask.tar.gz'
+    url_for_images = 'https://storage.googleapis.com/tz-ml-public/substation-over-10km2-csv-main-444e360fd2b6444b9018d509d0e4f36e/image_stack.tar.gz'
+    url_for_masks = 'https://storage.googleapis.com/tz-ml-public/substation-over-10km2-csv-main-444e360fd2b6444b9018d509d0e4f36e/mask.tar.gz'
+
+    def __init__(
+        self,
+        root: Path,
+        bands: Sequence[int],
+        mask_2d: bool,
+        timepoint_aggregation: str = 'concat',
+        download: bool = False,
+        checksum: bool = False,
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        num_of_timepoints: int = 4,
+        use_timepoints: bool = False,
+    ) -> None:
+        """Initialize the Substation.
+
+        Args:
+            root: Path to the directory containing the dataset.
+            bands: Number of channels to use from the image.
+            mask_2d: Whether to use a 2D mask.
+            timepoint_aggregation: How to aggregate multiple timepoints.
+            download: Whether to download the dataset if it is not found.
+            checksum: Whether to verify the dataset after downloading.
+            transforms: A transform takes input sample and returns a transformed version.
+            num_of_timepoints: Number of timepoints to use for each image.
+            use_timepoints: Whether to use multiple timepoints for each image.
+        """
+        self.root = root
+        self.bands = bands
+        self.mask_2d = mask_2d
+        self.timepoint_aggregation = timepoint_aggregation
+        self.download = download
+        self.use_timepoints = use_timepoints
+        self.checksum = checksum
+        self.transforms = transforms
+        self.image_dir = os.path.join(root, 'image_stack')
+        self.mask_dir = os.path.join(root, 'mask')
+        self.num_of_timepoints = num_of_timepoints
+        self._verify()
+        self.image_filenames = sorted(os.listdir(self.image_dir))
```

Comment:
We've seen reports that storing lists as dataset attributes results in memory leaks in Python multiprocessing. It may be better to store these as pandas Series.